code/game/objects/items/weapons/dnascrambler.dm

```diff
@@ -4,44 +4,37 @@
 	icon = 'icons/obj/syringe.dmi'
 	item_state = "syringe_0"
 	icon_state = "b10"
-	var/used = null
 
-	update_icon()
-		if(used)
-			icon_state = "b0"
+/obj/item/weapon/dnascrambler/attack(mob/M as mob, mob/user as mob)
+	if(!M || !user)
+		return
+
+	if(!ishuman(M) || !ishuman(user))
```

Comment:
Typecast the parameters and combine these and the above checks into `istype()`s.